modules/service/src/main/scala/lucuma/odb/service/GuideEnvironmentService.scala

```diff
@@ -0,0 +1,469 @@
+              shapeConstraint,
+              ags.widestConstraints.some
+            )
+          }
+          .toRight(
+            Error.GeneralError(
+              s"Unable to get tracking information for asterism for observation $oid."
+            )
+          )
+      }
+
+      def callGaia(
+        oid:     Observation.Id,
+        obsTime: Instant,
+        query:   ADQLQuery
+      ): F[Either[Error, NonEmptyList[GuideStarCandidate]]] = {
+        val MaxTargets                     = 100
+        given catalog: CatalogAdapter.Gaia = CatalogAdapter.Gaia3Lite
+        given ci: ADQLInterpreter          = ADQLInterpreter.nTarget(MaxTargets)
+        val request                        = Request[F](Method.GET,
+                                 CatalogSearch.gaiaSearchUri(query),
+                                 headers = Headers(("x-requested-with", "XMLHttpRequest"))
+        )
+        httpClient
+          .stream(request)
+          .flatMap(
+            _.body
+              .through(utf8.decode)
+              .through(CatalogSearch.guideStars[F](CatalogAdapter.Gaia3Lite))
+          )
+          .compile
+          .toList
+          .map(_.collect { case Right(s) => GuideStarCandidate.siderealTarget.get(s).at(obsTime) })
+          .map(
+            NonEmptyList
+              .fromList(_)
+              .toRight(
+                Error
+                  .GaiaError(s"No valid guide star candidates were returned for observation $oid.")
+              )
+          )
+          .handleError(e => Error.GaiaError(e.getMessage()).asLeft)
+      }
+
+      // TODO: use caching
+      def getCandidates(
+        oid:      Observation.Id,
+        obsTime:  Instant,
+        tracking: ObjectTracking
+      ): F[Either[Error, NonEmptyList[GuideStarCandidate]]] =
+        (for {
+          query      <- EitherT.fromEither(getGaiaQuery(oid, obsTime, tracking, probeArm.candidatesArea))
+          candidates <- EitherT(callGaia(oid, obsTime, query))
+        } yield candidates).value
+
+      extension [D](steps: NonEmptyList[Step[D]])
+        def offsets: List[Offset] = steps.collect { case Step(_, _, StepConfig.Science(offset, _), _, _, _) =>
+          offset
+        }
+
+      def offsetsFromExecConfig[S, D](
+        oid:        Observation.Id,
+        execConfig: ExecutionConfig[S, D]
+      ): Option[NonEmptyList[Offset]] =
+        execConfig.science
+          .map { s =>
+            s.nextAtom.steps.offsets ++
+              s.possibleFuture.flatMap(_.steps.offsets)
+          }
+          .flatMap(NonEmptyList.fromList)
+
+      def getOffsets(
+        pid: Program.Id,
+        oid: Observation.Id
+      ): F[Either[Error, Option[NonEmptyList[Offset]]]] =
+        generator(commitHash, itcClient, plannedTimeCalculator)
+          .digest(pid, oid)
+          .map {
+            _.leftMap(Error.GeneratorError(_))
+              .map { d => NonEmptyList.fromFoldable(d.science.offsets) }
```

Comment:
Do we need a union with acquisition offsets?  Seems like the guide star has to work for both.

Reply:
Good question. @cquiroz? I just stole this from explore.

Reply:
That's correct, we need to include acq offsets, Needs to be fixed in explore

Reply:
OK. I'll combine with the acq offsets.